fieldservice/models/fsm_order.py

```diff
@@ -225,18 +232,32 @@ def action_assign(self):
                                     "until 'Assigned To' is filled in"))
 
     def action_schedule(self):
-        return self.write({'stage_id': self.env.ref(
-            'fieldservice.fsm_stage_scheduled').id})
+        if self.scheduled_date_start and self.person_id:
+            return self.write({'stage_id': self.env.ref(
+                'fieldservice.fsm_stage_scheduled').id})
+        else:
+            raise ValidationError(_("Cannot move to Scheduled " +
+                                    "until both 'Assigned To' and " +
+                                    "'Scheduled Start Date' are filled in"))
 
     def action_enroute(self):
         return self.write({'stage_id': self.env.ref(
             'fieldservice.fsm_stage_enroute').id})
 
     def action_start(self):
+        if not self.date_start:
+            raise ValidationError(_("Cannot move to Start " +
+                                    "until 'Actual Start' is filled in"))
         return self.write({'stage_id': self.env.ref(
             'fieldservice.fsm_stage_started').id})
 
     def action_complete(self):
+        if not self.date_end:
+            raise ValidationError(_("Cannot move to Complete " +
+                                    "until 'Actual End' is filled in"))
+        if not self.resolution:
+            raise ValidationError(_("Cannot move to Complete " +
+                                    "until 'Resolution' is filled in"))
```

Comment:
@osi-scampbell This could be grouped into one if statement. If the user do not fill in those 2 fields, he will get the error message, fix the first field, try again and get the error message for the second field...